RevenueCatUI/Templates/TemplateViewType.swift

```diff
@@ -41,6 +41,11 @@ protocol TemplateViewType: SwiftUI.View {
     var configuration: TemplateViewConfiguration { get }
     var userInterfaceIdiom: UserInterfaceIdiom { get }
 
+    /// `UserInterfaceSizeClass` is only available for macOS/watchOS/tvOS with Xcode 15
+    #if swift(>=5.9) || (!os(macOS) && !os(watchOS) && !os(tvOS))
```

Comment:
Ideally we mark `TemplateViewType` as unavailable in these platforms (since it is), but that leads to a SwiftUI warning:
> _modify accessor cannot be more available than unavailable enclosing scope

And having a warning breaks CocoaPods linting.

Reply:
what's the story with visionOS for these? wondering if it'd be easier or safer to allow-list iOS rather than deny-list the rest

Reply:
It is available in visionOS.
I went with deny-list because presumedly we'll remove these eventually when adding support for them (other than watchOS which uses a separate template)